tests/components/test-components.php

```diff
@@ -47,4 +207,226 @@ public function test_template_default_context() {
 		$this->assertEquals( $post->ID, $context->get( 'irving/post_id' ), 'Default post ID context not set.' );
 		$this->assertEquals( new WP_Query( [] ), $context->get( 'irving/wp_query' ), 'Default wp query context not set.' );
 	}
+
+	/**
+	 * Test output for core components.
+	 *
+	 * @dataProvider get_core_component_data
+	 * @group core-components
+	 *
+	 * @param string   $name     Name of the component being tested.
+	 * @param callable $expected Callback to set the expected component after hydration.
+	 * @param callable $setup    Optional. A callback function used to set up state.
+	 */
+	public function test_core_components( string $name, callable $expected, callable $setup = null ) {
+		// Run setup logic needed before creating the component.
+		if ( ! empty( $setup ) ) {
+			call_user_func( $setup );
+		}
+
+		$expected  = call_user_func( $expected );
+		$component = new Component( $name );
+
+		$this->assertEquals(
+			$expected,
+			// Test in array syntax so it's easier to read diffs.
+			$component->to_array(),
+			sprintf( 'Broken component: %s', esc_html( $name ) )
+		);
+	}
+
+	/**
+	 * Data provider for test_core_components().
+	 *
+	 * @return array Test args
+	 */
+	public function get_core_component_data() {
+
+		return [
+			[
+				'irving/archive-title',
+				function () {
+					return [
+						'name'     => 'irving/archive-title',
+						'_alias'   => 'irving/text',
+						'config'   => (object) [
+							// The format of the archive title changed in WP version 5.5.
```

Comment:
What a pita.

Reply:
I could make 100 gyros with all this pita 😆